packages/compass-crud/src/components/delete-data-menu.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { Icon, Button, Tooltip } from '@mongodb-js/compass-components';
+import { usePreference } from 'compass-preferences-model';
+
+type DeleteMenuButtonProps = {
+  isWritable: boolean;
+  onClick: () => void;
+};
+
+const DeleteMenuButton: React.FunctionComponent<DeleteMenuButtonProps> = ({
+  isWritable,
+  onClick,
+}) => {
+  const isVisible = usePreference('enableBulkDeleteOperations', React);
+
+  if (!isVisible) {
+    return <></>;
```

Comment:
Shouldn't this just be `return null;` I'm no expert, but I think that's what's idiomatic here.